### PR TITLE
feat: Add IsParseErr helper(issue 559)

### PR DESCRIPTION
**Pull request by LWF15**
add the global variable call ErrParse and set it to the struct of RedisMessage as function IsParseErr and add some judgment condition in some toXX() type method in PR555

### CODE REVIEW

**Comment by LWF15** [comment]
by the way, sorry about the title, I remember the wrong issue I working with Q A Q

File: message.go [hunk 1]
```diff
@@ -569,6 +579,9 @@ func (m *RedisMessage) ToString() (val string, err error) {
 	if m.IsString() {
 		return m.string, nil
 	}
+	if m.IsParseErr() {
+		return "", fmt.Errorf("%w: redis message type %s is not a string", errParse, typeNames[m.typ])
+	}
 	if m.IsInt64() || m.values != nil {
 		typ := m.typ
 		return "", fmt.Errorf("redis message type %s is not a string", typeNames[typ])
```
[the remark this answers is not in the document]
Comment (LWF15) [reply message.go:587]: ok, I will change it!

File: message.go [hunk 2]
```diff
@@ -520,6 +525,11 @@ func (m *RedisMessage) IsNil() bool {
 	return m.typ == typeNull
 }
 
+// IsParseErr checks if the error is a parse error
+func (m *RedisMessage) IsParseErr() bool {
```
[the remark this answers is not in the document]
Comment (LWF15) [reply message.go:529]: Ok, I would try to fix it

**Comment by rueian** [comment]
> by the way, sorry about the title, I remember the wrong issue I working with Q A Q

No worry. Please also write a test for the `IsParseErr`.

**Comment by LWF15** [comment]
> > by the way, sorry about the title, I remember the wrong issue I working with Q A Q
> 
> No worry. Please also write a test for the `IsParseErr`.

Ok, I will wrote it

**Comment by rueian** [comment]
Merged. Thank you for your contributions @LWF15!

**Pull request merged**